### PR TITLE
Forbid usage of kotlin-android-extensions

### PR DESCRIPTION
**Pull request by aitorvs**
Task/Issue URL: https://app.asana.com/0/488551667048375/1202728801491359/f

### Description
Forbid the usage of `kotlin-android-extension` as it is deprecated.

We use(d) it for two things:
* `@Parcelize`
* synthetics

For some modules like `:app` and `:vpn-impl` I've added an exception
because we first need to migrate way from synthetics

### Steps to test this PR
Smoke tests for autofill

### CODE REVIEW

**Comment by aitorvs** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #2282** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/2282" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/2282?utm_source=stack-comment).

File: autofill/autofill-api/build.gradle [hunk 1]
```diff
@@ -17,7 +17,7 @@
 plugins {
     id 'com.android.library'
     id 'kotlin-android'
-    id 'kotlin-android-extensions'
+    id 'kotlin-parcelize'
```
Comment (CDRussell) [inline autofill/autofill-api/build.gradle:20]: 👍

File: ad-click/ad-click-impl/build.gradle [hunk 2]
```diff
@@ -18,7 +18,6 @@ plugins {
     id 'com.android.library'
     id 'kotlin-android'
     id 'com.squareup.anvil'
-    id 'kotlin-android-extensions'
```
Comment (anikiki) [inline ad-click/ad-click-impl/build.gradle:21]: :+1:

**Review by CDRussell — APPROVED** [review]

**Comment by CDRussell** [comment]
Autofill functionality all still good 👍

**Pull request merged**